src/inputs/EditableLabel/EditableLabel.js

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { TextField, makeStyles, Typography, CircularProgress } from '@material-ui/core';
+import EditIcon from '@material-ui/icons/Edit';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    height: '100%',
+  },
+  labelDisplayContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'stretch',
+  },
+  label: {
+    maxWidth: '500px',
+    color: theme.palette.primary.main,
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+  textField: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'stretch',
+    '& .MuiOutlinedInput-input': {
+      borderWidth: '2px',
+      outline: 'none',
+      padding: '8px',
+    },
+    '& .MuiOutlinedInput-root': {
+      minWidth: '300px',
+    },
+  },
+  editIcon: {
+    marginRight: '10px',
+    fontWeight: 'bold',
+    height: '20px',
+    color: theme.palette.primary.main,
+  },
+  circularLoading: {
+    marginLeft: '15px',
+    color: theme.palette.primary.contrastText,
+    backgroundColor: theme.palette.background.card,
+  },
+}));
+
+export const EditableLabel = (props) => {
+  const { value, onNewValue, checkValue, labels, ...typographyprops } = props;
+
+  const classes = useStyles();
+
+  const [isEditing, setEditing] = useState(false);
+  const [error, setError] = useState(null);
+  const [isHovered, setHovered] = useState(false);
+  const [textFieldValue, setTextFieldValue] = useState(value);
+
+  const startEdition = (event) => {
+    event.stopPropagation();
+    event.preventDefault();
+    setEditing(true);
+  };
+
+  const checkNewValueOnChange = (event) => {
+    const newValue = event.target.value;
+    if (newValue !== value) {
+      setError(checkValue(newValue));
+    }
+  };
+
+  const stopEdition = (event) => {
+    event.stopPropagation();
+    const newValue = event.target.value;
+    if (newValue !== value) {
+      if (error == null) {
+        onNewValue(newValue);
+        setTextFieldValue(newValue);
+      }
+    }
+    setEditing(false);
+    setHovered(false);
+  };
+
+  const getLabelDisplay = () => {
+    if (value === textFieldValue) {
+      return (
+        <div
+          className={classes.labelDisplayContainer}
+          onMouseLeave={() => setHovered(false)}
+          onMouseOver={() => setHovered(true)}
+          onClick={startEdition}
+        >
+          {isHovered && <EditIcon className={classes.editIcon} />}
+          <Typography className={classes.label} variant={typographyprops.variant}>
+            {value}
+          </Typography>
+        </div>
+      );
+    } else {
+      return <CircularProgress size={25} className={classes.circularLoading} />;
+    }
+  };
+
+  return (
+    <div data-cy="editable-label" id="EditableLabel">
```

Comment:
The `id` value should probably not start with a capital letter.
Also, do we really need an `id` for this div ?

Reply:
I remove it